lpcompiler.pas

```diff
@@ -1342,12 +1344,17 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
         begin
           if addToScope then
             FStackInfo.addSelfVar(Lape_SelfParam, TLapeType(Typ));
+
+          methodDef := Result.MethodDef;
           Result.Free();
           Result := TLapeType_MethodOfType.Create(Self, TLapeType(Typ), nil, nil, '', @Pos);
+          Result.MethodDef := methodDef;
         end
         else if (not (Typ is TLapeType_SystemUnit)) then
           LapeException(lpeTypeExpected, [Tokenizer]);
-      end;
+      end
+      else if (Result.MethodDef = mdProperty) then
+        LapeException(lpeExpectedProperty, Tokenizer.DocPos);
```

Comment:
I guess you can make the same argument for properties of types. But okay, let's use it for types only.